pkg/testcase/conformance.go

```diff
@@ -0,0 +1,137 @@
+package testcase
+
+import (
+	"strings"
+
+	"github.com/rancher/distros-test-framework/pkg/customflag"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// TestSonobuoyMixedOS runs sonobuoy tests for mixed os cluster (linux + windows) node.
+func TestSonobuoyMixedOS(deleteWorkload bool) {
+	installConformanceBinary()
+
+	cmd := "sonobuoy run --kubeconfig=" + shared.KubeConfigFile +
+		" --plugin my-sonobuoy-plugins/mixed-workload-e2e/mixed-workload-e2e.yaml" +
+		" --aggregator-node-selector kubernetes.io/os:linux --wait"
+	res, err := shared.RunCommandHost(cmd)
+	Expect(err).NotTo(HaveOccurred(), "failed output: "+res)
+
+	cmd = "sonobuoy retrieve --kubeconfig=" + shared.KubeConfigFile
+	testResultTar, err := shared.RunCommandHost(cmd)
+	Expect(err).NotTo(HaveOccurred(), "failed cmd: "+cmd)
+
+	cmd = "sonobuoy results  " + testResultTar
+	res, err = shared.RunCommandHost(cmd)
+	Expect(err).NotTo(HaveOccurred(), "failed cmd: "+cmd)
+	Expect(res).Should(ContainSubstring("Plugin: mixed-workload-e2e\nStatus: passed\n"))
+
+	if deleteWorkload {
+		cmd = "sonobuoy delete --all --wait --kubeconfig=" + shared.KubeConfigFile
+		_, err = shared.RunCommandHost(cmd)
+		Expect(err).NotTo(HaveOccurred(), "failed cmd: "+cmd)
+		sonobuoyVersion := customflag.ServiceFlag.External.SonobuoyVersion
+		err = shared.InstallSonobuoy("delete", sonobuoyVersion)
+		if err != nil {
+			GinkgoT().Errorf("error: %v", err)
+			return
+		}
+	}
+}
+
+func ConformanceTest(cluster *shared.Cluster) {
+	verifyClusterNodes(cluster)
+	installConformanceBinary()
+	launchSonobuoyTests("certified-conformance")
+	// refactor later to accept quick or certified-conformance
+	testResultTar := checkStatusGetResults(cluster)
+	shared.LogLevel("info", "%s", "testResultTar: "+testResultTar)
+	// need to do cilium force failures to test
+	rerunFailedTests(testResultTar)
+	parseResults(testResultTar)
+	cleanupTests()
+}
+
+func verifyClusterNodes(cluster *shared.Cluster) bool {
+	if cluster.NumAgents < 1 && cluster.NumServers < 1 {
+		shared.LogLevel("error", "%s", "cluster does not meet the minimum requirements to run conformance tests")
+		return false
+	}
+
+	return true
+}
+
+func installConformanceBinary() {
+	sonobuoyVersion := customflag.ServiceFlag.External.SonobuoyVersion
+	err := shared.InstallSonobuoy("install", sonobuoyVersion)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func launchSonobuoyTests(testMode string) {
+	// not doing anything different yet if the status is running from the previous attempts
+	cmds := "kubectl get namespace sonobuoy --kubeconfig=" + shared.KubeConfigFile
```

Comment:
nit: use shorthand `namespace` -> `-n`
Not sure what the comment means, can you clarify?